lib/npg/email/run.pm

```diff
@@ -72,36 +64,34 @@ sub study_lane_followers {
   my ($self, $id_run) = @_;
 
   my $return_hash = {};
-  eval {
+  try {
     my $lims = $self->get_lims($id_run);
     my $with_spiked_control = 0;
     foreach my $child_lims ($lims->associated_child_lims) {
       my $position = $child_lims->position;
       foreach my $study_name ($child_lims->study_names($with_spiked_control)) {
         if ( !exists $return_hash->{ $study_name } ) {
-          $return_hash->{$study_name}->{followers} = undef;
-          $return_hash->{$study_name}->{lanes} = [];
+          $return_hash->{$study_name}->{'followers'} = undef;
+          $return_hash->{$study_name}->{'lanes'} = [];
         }
-        push @{ $return_hash->{$study_name}->{lanes} }, {position => $position,library => $child_lims->library_name,};
+        push @{ $return_hash->{$study_name}->{'lanes'} },
+          {'position' => $position,'library' => $child_lims->library_name,};
```

Comment:
On 23/10/2015 09:18, dkj wrote:
>
> In lib/npg/email/run.pm 
> <https://github.com/wtsi-npg/npg_tracking/pull/305#discussion_r42842343>:
>
> >          }
> > -        push @{ $return_hash->{$study_name}->{lanes} }, {position => $position,library => $child_lims->library_name,};
> > +        push @{ $return_hash->{$study_name}->{'lanes'} },
> > +          {'position' => $position,'library' => $child_lims->library_name,};
>
> Potentially same problem here?
>

Yes, comes up in a different cron.  Just pushed the fix. There is also 
another curious warning

Use of uninitialized value in ucfirst at/software/npg/20151019/bin/../lib/perl5/npg/email/event/status_change/run.pm line 168.

which suggests that an attempt to process incomplete or errored outcomes is made. Looking at at now

> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/wtsi-npg/npg_tracking/pull/305/files#r42842343>.
>




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.